docs/spec/spec-proposals/f1-fee-distribution/f1_fee_distr.tex

```diff
@@ -238,11 +234,11 @@ \section{TO DOs}
 
 \begin{itemize}
 	\item A global fee pool can be described.
-	\item Determine if auto-bonding fees is compatible with inflation and comission rates in conjunction
-	\item mention storage optimization in the uniform inflation and iteration over slashing case: only have one storage location w/ refcount of the product of inflation results
-	\item Remove iteration over slashes by the same normalization trick used in inflation
+	\item Mention storage optimization for how to prune slashing entries in the uniform inflation and iteration over slashing case
 	\item Add equation numbers
 	\item perhaps re-organize so that the no iteration
+ 	\item Section on decimal precision considerations (would unums help?), and mitigating errors in calculation with floats and decimals. -- This probably belongs in a corrollary markdown file in the implementation
```

Comment:
This would be helpful - what do you mean by unums? We definitely can't use floating point types.

I'm already a bit concerned with loss of precision in the current F1 implementation (which does not include inflation).

Reply:
Unums are an alternative to floating points. They have claim to have more desirable properties around error propogation, domain usage, intervals, and energy efficiency. (There are lots of weird properties with floats)

Floating points may be usable. The concern that exists is "IEEE 754" standard isn't fully met, so you get issues with errors on the order of 1 or 2 ULPs, denorms, and NANs, but those can be taken care of in software. (And a virtual IEEE 754 used in the cases where there isn't hardware support) It may be a footgun, but probably worth seeing if errors are mystically negligible there. (I do doubt that this is the case)

Reply:
I imagine unums are a pretty bad idea without hardware support. This really doesn't fit into the ideal usage of floats, so I imagine decimal is the way to go.

Reply:
Unums are cool! But probably too experimental for our purposes right now.

I don't know if we've done a thorough investigation of floating-point. The "blockchain precedent" certainly seems negative (I can't think of any counterexamples, can you?) - which doesn't necessarily portend impossibility, but it's a signal against in absence of detailed research ourselves.

Reply:
I'm looking forward to unums as well!

Agreed w.r.t. to floats. I don't think we really fit into the ideal usage of floats, since we do want high precision.